src/ui/RadarMap.ts

```diff
@@ -69,22 +72,24 @@ class RadarMap extends maplibregl.Map {
 
     // configure maplibre options
     const style = getStyle(config, mapOptions);
-    const maplibreOptions: maplibregl.MapOptions = Object.assign({},
+    const maplibreOptions: RadarMapOptions = Object.assign({},
       defaultMaplibreOptions,
       mapOptions,
       { style },
     );
     Logger.debug(`initialize map with options: ${JSON.stringify(maplibreOptions)}`);
 
-    maplibreOptions.transformRequest = (url, resourceType) => {
+    (maplibreOptions as maplibregl.MapOptions).transformRequest = (url, resourceType) => {
+      // this handles when a style is switched
       if (resourceType === 'Style' && isRadarStyle(url)) {
-        url = createStyleURL(config, url);
+        url = createStyleURL(config, { ...maplibreOptions, style: url });
       }
 
       let headers = {
         'Authorization': config.publishableKey,
         'X-Radar-Device-Type': 'Web',
         'X-Radar-SDK-Version': SDK_VERSION,
+        'X-Radar-Device-Language': navigator.language,
```

Comment:
there's also `navigator.languages` which returns an array of the browser's selected languages.  we can use those to order fallback language priority. e.g. `[fr-CA, en-CA]` (candian french fallback to canadian english)

Reply:
It might make sense to move this into `sr/navigator` for consistently, and maybe handle the case where navigator is not available.

Reply:
ah right forgot that existed

Reply:
done

Reply:
just returns `undefined` if navigator isn't available